core/src/com/unciv/logic/civilization/TechManager.kt

```diff
@@ -165,6 +167,16 @@ class TechManager {
         val currentTechnology = currentTechnologyName()
         if (currentTechnology == null) return
         techsInProgress[currentTechnology] = researchOfTech(currentTechnology) + scienceForNewTurn
+        if (getScienceFromResearchAgreement!=0){ // https://forums.civfanatics.com/resources/research-agreements-bnw.25568/
+            var researchAgreementModifier= 0.5f
+            if (civInfo.policies.isAdopted("Scientific Revolution"))
+                researchAgreementModifier += 0.25f
+            if (civInfo.containsBuildingUnique("Science gained from research agreements +50%"))
+                researchAgreementModifier += 0.25f
+            getScienceFromResearchAgreement = (getScienceFromResearchAgreement / 3 * researchAgreementModifier).toInt()
+            techsInProgress[currentTechnology] = techsInProgress[currentTechnology]!! + getScienceFromResearchAgreement
+            getScienceFromResearchAgreement = 0
+        }
```

Comment:
I think this should be its own little function, getScienceFromResearchAgreements()

Reply:
Using getScienceFromResearchAgreements() in code may lead to declaration error, so i delete the last "s". of course above this, "scienceFromResearchAgreements" does't include "s" too.

Reply:
"getScienceFromResearchAgreements()" has the word "scienceFromResearchAgreements" in its last leads to declaration error. you can try this.